File: techila/examples/C/Tutorial/1_distribution/local_program.c
// Copyright 2011-2013 Techila Technologies Ltd.

/* This file contains the locally executable program and does not communicate
with the Techila environment in any way. This program will be compiled when
passing the argument 'local_program' to make. */

#include <stdio.h>

int main(int argc, char** argv)
{
  int loops = 5;  // Number of iterations in the for-loop
  int result;
  int i;
  int myarray[5]; // Array for the results

  for (i = 0; i < loops; i++) {
    result = 1 + 1;  // Simple arithmetic operation
    myarray[i] = result; // Store the result in array
    printf("Iteration %d, Result: %d\n", i + 1, myarray[i]); // Display progress
  }

  // Display the contents of the array
  printf("Content of the array 'myarray': %i %i %i %i %i\n",
         myarray[0],
         myarray[1],
         myarray[2],
         myarray[3],
         myarray[4]);
  return 0;
}

File: techila/examples/CSharp/Tutorial/1_distribution/LocalProgram.cs
using System;
using System.Collections.Generic;


namespace LocalProgram
{
    class LocalProgram
    {
        static void Main()
        {
            // Set 'iterations' to 5. Will be used to define the number of iterations later.
            int iterations = 5;

            // Do 5 iterations.
            for (int i = 0; i < iterations; i++)
            {
                // Create new instance of 'DistributionDist'
                DistributionDist dd = new DistributionDist();

                // Call the 'Execute' method and store result variable in 'res'
                int res = dd.Execute();

                // Print the value of the 'res' variable.
                System.Console.WriteLine("Result of iteration {0}: {1}",i + 1, res);
            }


        }



    }
    class DistributionDist // Class containing some executable code
    {
        // Define type of variable that will be returned
        int Result;

        // Define an 'Execute' method containing the code that will be executed
        public int Execute()
        {
            // Do simple arithmetic operation
            Result = 1 + 1;

            // Return the result
            return (Result);
        }

    }
}
